### PR TITLE
Use leveldb to instead boltdb on sqlchain

### PR DESCRIPTION
**Pull request by zeqing-guo**
Use leveldb to instead boltdb on sqlchain

### CODE REVIEW

File: sqlchain/chain.go [hunk 1]
```diff
@@ -374,21 +364,25 @@ func (c *Chain) pushBlock(b *ct.Block) (err error) {
...
-		}
-
-		c.rt.setHead(st)
-		c.bi.addBlock(node)
-		c.qi.setSignedBlock(h, b)
+	batch := new(leveldb.Batch)
+	batch.Put(metaState[:], encState.Bytes())
+	t, err := c.bdb.OpenTransaction()
+	if err = t.Put(metaState[:], encState.Bytes(), nil); err != nil {
+		err = errors.Wrapf(err, "put %s", string(metaState[:]))
```
Comment (leventeliu) [inline sqlchain/chain.go:369]: Need to rollback here?

File: utils/bytes.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+
+package utils
+
+import "bytes"
+
+// ConcatAll concatenate several bytes slice into one.
+func ConcatAll(args ...[]byte) []byte {
+	key := bytes.NewBuffer([]byte{})
```
Comment (leventeliu) [inline utils/bytes.go]: Use pre-allocated slice with the total length and the builtin append will be more efficient.

File: cmd/cql-minerd/main.go [hunk 3]
```diff
@@ -120,7 +120,7 @@ func initLogs() {
 func main() {
 	// set random
 	rand.Seed(time.Now().UnixNano())
-	log.SetLevel(log.InfoLevel)
+	log.SetLevel(log.DebugLevel)
```
Comment (auxten) [inline cmd/cql-minerd/main.go]: Log level should not be the debug level for PR

File: utils/bytes.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+// ConcatAll concatenate several bytes slice into one.
+func ConcatAll(args ...[]byte) []byte {
+	var bLen int
+	for i := range args {
+		bLen += len(args[i])
+	}
+
+	key := make([]byte, bLen)
+	bLen = 0
```
Comment (auxten) [inline utils/bytes.go]: don't reuse var

**Review by auxten — APPROVED** [review]
LGTM

**Pull request merged**